demo/src/App.vue

```diff
@@ -20,7 +20,16 @@
       :showTypingIndicator="showTypingIndicator"
       :titleImageUrl="titleImageUrl"
       @onType="handleOnType"
-    />
+    >
+      <template v-slot:header>
+        🤔 Good chat with {{participants.map(m=>m.name).join(' & ')}}
+      </template>
+      <template v-slot:text-message-body="{ message }">
+        <small style="background:red" v-if="message.meta">{{message.meta}}</small>{{message.text}}
```

Comment:
why are the metas in red background?
why are they at the beginning of the message?

Reply:
why did you remove message formatting?

Reply:
Slot is good for custom format of messages. If you would like to Format DateTime with custom locale, it is not possible in current state. Or for example add delete/edit buttons, and 'edited' icon.

Also slot for avatar is needed for me, as on my site we using imageUrl for that, and that's ok, but if user didn't selected, we render Colored Round with First letters of name (vue-avatar), and that case is not possible in current chat.

Custom styles on demo, is only demo :) If you remove slots from App.vue, it will be rendered with default - like it was before.

I'll remove that from Demo project, sorry.

Reply:
![изображение](https://user-images.githubusercontent.com/5583742/59381076-4cb94600-8d63-11e9-971d-16094574d368.png)

like that (custom datetime), avatars not only images, and I need avatar on header :)